servicetalk-concurrent-api/src/testFixtures/java/io/servicetalk/concurrent/api/TestExecutor.java

```diff
@@ -314,8 +317,14 @@ private static boolean executeOne(Queue<RunnableWrapper> tasks, AtomicInteger ta
         return false;
     }
 
-    // Wraps Runnables to ensure that object-equality (and hashcode) is used for removal from Lists.
-    // Also ensures a unique object each time, so the same Runnable can be executed multiple times.
+    /**
+     *  Wraps Runnables to ensure that object-equality (and hashcode) is used for removal from Lists.
+     *  Also ensures a unique object each time, so the same Runnable can be executed multiple times.
+     *  Sets the thread name to {@code TestExecutor} while running the task so that capturing the thread name makes
+     *  sense and during debugging the execution context is more obvious.
+     *  Pessimistically set the {@link AsyncContextMap} to null to ensure that any use of {@link AsyncContextMap}
+     *  within the context of the Runnable includes appropriate setting/restoring.
```

Comment:
I'm not sure about this change. We expect `AsyncContext` to be propagated through the whole reactive chain, regardless of on which executor each operator is executed. Will this change the behavior? Can we reuse `ContextPreservingRunnable` in this case to be consistent with other executors?

Reply:
The intent of this change is to clear the current async context from the thread which will execute the `Runnable` to ensure that if the `Runnable` does depend upon `AsyncContext` it is properly wrapped in `ContextPreservingRunnable`. It is an "adversarial" testing behaviour intended to simulate what an offloaded task would experience if it were run on a fresh thread. Alternately we could also add a new `AsyncContextMap` implementation which asserts if any of the `get()`/`put()` methods are called rather than (potentially) a new `AsyncContextMap` being created as happens under some scenarios.